node.go

```diff
@@ -38,13 +38,71 @@ type Node struct {
 	Type          string                 `json:"type"`
 	ID            string                 `json:"id,omitempty"`
 	ClientID      string                 `json:"client-id,omitempty"`
-	Attributes    map[string]interface{} `json:"attributes,omitempty"`
+	Attributes    attributes             `json:"attributes,omitempty"`
 	Relationships map[string]interface{} `json:"relationships,omitempty"`
 	Links         *Links                 `json:"links,omitempty"`
 	Meta          *Meta                  `json:"meta,omitempty"`
 }
 
+func (n *Node) handleNodeErrors() {
+	for k, v := range n.Attributes {
+		if _, ok := v.(nodeError); ok {
+			delete(n.Attributes, k)
+		}
+	}
+}
+
+type attributes map[string]interface{}
+
+// this attributes setter
+// * adds new entries as-is
+// * tracks dominant field conflicts
+func (a attributes) set(k string, v interface{}) {
+	if val, ok := a[k]; ok {
+		if ne, ok := val.(nodeError); ok {
+			ne.Add(k, v)
+			return
+		}
+		a[k] = newDominantFieldConflict(k, val, v)
+	} else {
+		a[k] = v
+	}
+}
+
+// mergeAttributes consolidates 2 attribute maps
+// if there are conflicting keys, the values in the argument take priority
+func (n *Node) mergeAttributes(attrs attributes) {
+	for k, v := range attrs {
+		n.Attributes[k] = v
+	}
+}
+
+// mergeAttributes consolidates multiple nodes into a single consolidated node
```

Comment:
This comment doesn't seem to make sense here: seems to be for a different function.